File: endpoints-under-test/dart-dart_frog/lib/model/note.dart
class Note {
  final int id;
  final String noteText;
  final int applicationId;

  Note({
    required this.id,
    required this.noteText,
    required this.applicationId,
  });

  static Note? fromMap(Map<String, dynamic>? map) {
    if (map == null ||
        map['id'] == null ||
        map['application_id'] == null ||
        map['note_text'] == null) {
      return null;
    }

    return Note(
      id: map['id'] as int,
      noteText: map['note_text'] as String,
      applicationId: map['application_id'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'noteText': noteText,
      'applicationId': applicationId,
    };
  }
}

File: app/model/note.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from app.config import Base

class Note(Base):
    __tablename__ = "notes"

    id = Column(Integer, primary_key=True, index=True)
    noteText = Column('note_text', String)
    applicationId = Column('application_id', Integer, ForeignKey('applications.id'))

    application = relationship("Application", back_populates="notes")
